src/parsers/archaea_parsers.py

```diff
@@ -0,0 +1,564 @@
+                if db_type == "PubMed":
+                    publications.append(f"PMID:{db_id}")
+                elif db_type == "DOI":
+                    publications.append(f"DOI:{db_id}")
+                elif db_type in ["EMBL", "GenBank", "DDBJ"]:
+                    publications.append(f"{db_type}:{db_id}")
+
+    return publications
+
+
+def parse_uniprot_entry(entry):
+    """
+    Parse a single <entry> element from a UniProt XML file into structured CDM-compatible records.
+
+    This function orchestrates parsing across different UniProt XML substructures, extracting
+    information such as entity metadata, identifiers, protein names, sequence-level, associations, 
+    and linked publications.
+
+    Args:
+        entry (xml.etree.ElementTree.Element): The UniProt <entry> XML element.
+
+    Returns:
+        dict: A dictionary containing extracted records across CDM tables:
+              - 'entity': core metadata
+              - 'identifiers': UniProt accessions
+              - 'names': protein naming data
+              - 'protein': sequence and EC info
+              - 'associations': taxonomy, cross-references, evidence-backed links
+              - 'publications': linked literature
+    """
+    ## Generate a globally unique CDM entity ID (UUID-based)
+    cdm_id = generate_unique_id()
+
+    # Build the core entity record for the protein, capturing creation/modification metadata
+    entity = {
+        "entity_id": cdm_id, 
+        "entity_type": "protein", # CDM classification
+        "data_source": "UniProt archaea", # Provenance
+        "created": entry.attrib.get("created"), # Date entry was first added
+        "updated": entry.attrib.get("modified") or entry.attrib.get("updated"), # Last modified
+        "version": entry.attrib.get("version"), # UniProt version number
+    }
+    
+    # Parse all <evidence> tags into a mapping from evidence keys to metadata
+    evidence_map = parse_evidence_map(entry)
+    
+    # Return all extracted components organized under CDM table keys
+    return {
+        "entity": entity,
+        "identifiers": parse_identifiers(entry, cdm_id),
+        "names": parse_names(entry, cdm_id),
+        "protein": parse_protein_info(entry, cdm_id),
+        "associations": parse_associations(entry, cdm_id, evidence_map),
+        "publications": parse_publications(entry),
+    }
+
+
+def stream_uniprot_xml_gz(url):
+    """
+    Stream and parse a UniProt XML .gz file from a remote URL.
+
+    This function uses streaming HTTP and on-the-fly GZIP decompression to efficiently
+    iterate over large UniProt XML files without loading the entire content into memory.
+
+    Args:
+        url (str): URL to the gzipped UniProt XML file.
+
+    Yields:
+        xml.etree.ElementTree.Element: Each complete <entry> element in the XML file.
+
+    """
+
+    # Make a streaming HTTP GET request to avoid loading the whole file
+    with requests.get(url, stream=True) as r:
```

Comment:
Given the size of the file and the possibility of broken connections, it's probably better to download it to the cdm server and then stream it from the local file.